modin/core/storage_formats/pandas/parsers.py

```diff
@@ -564,12 +564,13 @@ def update_row_nums(match):
             has_index_names=is_list_like(header) and len(header) > 1,
             skiprows=skiprows,
             usecols=usecols,
+            skip_blank_lines=False,
             **kwargs,
         )
-        # In excel if you create a row with only a border (no values), this parser will
-        # interpret that as a row of NaN values. pandas discards these values, so we
-        # also must discard these values.
-        pandas_df = parser.read().dropna(how="all")
+        pandas_df = parser.read()
+        if len(pandas_df) > 1 and pandas_df.isnull().all().all():
```

Comment:
Could you please add a comment explaining why we create a new pandas DataFrame if this test is true?